adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java

```diff
@@ -280,15 +310,30 @@ private String getCountryCode(CartData cartData) {
 
 
     public CaptureRequest createCaptureRequest(final String merchantAccount, final BigDecimal amount, final Currency currency, final String authReference, final String merchantReference) {
-        return new CaptureRequest().fillAmount(String.valueOf(amount), currency.getCurrencyCode()).merchantAccount(merchantAccount).originalReference(authReference).reference(merchantReference);
+        CaptureRequest request = new CaptureRequest().fillAmount(String.valueOf(amount), currency.getCurrencyCode())
+                                                     .merchantAccount(merchantAccount)
+                                                     .originalReference(authReference)
+                                                     .reference(merchantReference);
+        ApplicationInfo applicationInfo = updateApplicationInfo(request.getApplicationInfo());
+        request.setApplicationInfo(applicationInfo);
+        return request;
     }
 
     public CancelOrRefundRequest createCancelOrRefundRequest(final String merchantAccount, final String authReference, final String merchantReference) {
-        return new CancelOrRefundRequest().merchantAccount(merchantAccount).originalReference(authReference).reference(merchantReference);
+        CancelOrRefundRequest request = new CancelOrRefundRequest().merchantAccount(merchantAccount).originalReference(authReference).reference(merchantReference);
+        ApplicationInfo applicationInfo = updateApplicationInfo(request.getApplicationInfo());
+        request.setApplicationInfo(applicationInfo);
+        return request;
     }
 
     public RefundRequest createRefundRequest(final String merchantAccount, final BigDecimal amount, final Currency currency, final String authReference, final String merchantReference) {
-        return new RefundRequest().fillAmount(String.valueOf(amount), currency.getCurrencyCode()).merchantAccount(merchantAccount).originalReference(authReference).reference(merchantReference);
+        RefundRequest request = new RefundRequest().fillAmount(String.valueOf(amount), currency.getCurrencyCode())
+                                                   .merchantAccount(merchantAccount)
+                                                   .originalReference(authReference)
+                                                   .reference(merchantReference);
+        ApplicationInfo applicationInfo = updateApplicationInfo(request.getApplicationInfo());
+        request.setApplicationInfo(applicationInfo);
```

Comment:
shall we merge the setApplicationInfo into the new RefundRequest and merge it so we don't have to store it in a variable ?

Reply:
ok let's keep it as it is.